internal/storage/bucket_handle.go

```diff
@@ -477,14 +477,23 @@ func (b *bucketHandle) DeleteFolder(ctx context.Context, folderName string) (err
 	return err
 }
 
-func (b *bucketHandle) RenameFolder(ctx context.Context, folderName string, destinationFolderId string) (resp *control.RenameFolderOperation, err error) {
+func (b *bucketHandle) RenameFolder(ctx context.Context, folderName string, destinationFolderId string) (folder *gcs.Folder, err error) {
+	var controlFolder *controlpb.Folder
 	req := &controlpb.RenameFolderRequest{
 		Name:                "projects/_/buckets/" + b.bucketName + "/folders/" + folderName,
 		DestinationFolderId: destinationFolderId,
 	}
-	resp, err = b.controlClient.RenameFolder(ctx, req)
+	resp, err := b.controlClient.RenameFolder(ctx, req)
+	if err != nil {
+		return nil, fmt.Errorf("longrunning operation: %w", err)
```

Comment:
nit: what does appennding longrunning operation mean here?

Reply:
There were two error handling scenarios: one for when RenameFolder received a long-running operation and another after fetching the full response. I added code to differentiate between these cases. 
However, since the Go SDK handles returning the appropriate errors, I removed the custom error handling.

Done